target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4018-jalapeno.dts

```diff
@@ -0,0 +1,277 @@
+				reg = <0>;
+				compatible = "jedec,spi-nor";
+				spi-max-frequency = <24000000>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					partition0@0 {
+						label = "SBL1";
+						reg = <0x00000000 0x00040000>;
+						read-only;
+					};
+					partition1@40000 {
+						label = "MIBIB";
+						reg = <0x00040000 0x00020000>;
+						read-only;
+					};
+					partition2@60000 {
+						label = "QSEE";
+						reg = <0x00060000 0x00060000>;
+						read-only;
+					};
+					partition3@c0000 {
+						label = "CDT";
+						reg = <0x000c0000 0x00010000>;
+						read-only;
+					};
+					partition4@d0000 {
+						label = "DDRPARAMS";
+						reg = <0x000d0000 0x00010000>;
+						read-only;
+					};
+					partition5@e0000 {
+						label = "APPSBLENV"; /* uboot env*/
+						reg = <0x000e0000 0x00010000>;
+						read-only;
+					};
+					partition5@f0000 {
+						label = "APPSBL"; /* uboot */
+						reg = <0x000f0000 0x00080000>;
+						read-only;
+					};
+					partition5@170000 {
+						label = "ART";
+						reg = <0x00170000 0x00010000>;
+						read-only;
+					};
+				};
+			};
+
+			mt29f@1 {
+				status = "okay";
+				#address-cells = <1>;
+				#size-cells = <0>;
+				compatible = "spinand,mt29f", "w25n01gv";
+				reg = <1>;
+				spi-max-frequency = <24000000>;
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					partition0@0 {
+						label = "ubi";
+						reg = <0x00000000 0x08000000>;
+					};
+				};
+			};
+		};
+
+		usb3: usb3@8af8800 {
+			status = "okay";
+		};
+
+		crypto@8e3a000 {
+			status = "okay";
+		};
+
+		watchdog@b017000 {
+			status = "okay";
+		};
+
+		ess-switch@c000000 {
+			status = "okay";
+			switch_lan_bmp = <0x10>; /* lan port bitmap */
+		};
+
+		edma@c080000 {
+			status = "okay";
+		};
+	};
+};
+
+&tlmm {
+	mdio_pins: mdio_pinmux {
+		pinmux_1 {
+			pins = "gpio53";
+			function = "mdio1";
+		};
+		pinmux_2 {
+			pins = "gpio52";
+			function = "mdc";
+		};
+		pinconf {
+			pins = "gpio52", "gpio53";
+			bias-pull-up;
+		};
+	};
+
+	serial_pins: serial_pinmux {
+		mux {
+			pins = "gpio60", "gpio61";
+			function = "blsp_uart0";
+			bias-disable;
+		};
+	};
+
+	spi_0_pins: spi_0_pinmux {
+		pin {
+			function = "blsp_spi0";
+			pins = "gpio55", "gpio56", "gpio57";
+			drive-strength = <2>;
+			bias-disable;
+		};
+		pin_cs {
+			function = "gpio";
+			pins = "gpio54", "gpio59";
+			drive-strength = <2>;
+			bias-disable;
+			output-high;
+		};
+	};
+};
+
+&blsp_dma {
+	status = "okay";
+};
+
+&cryptobam {
+	status = "okay";
+};
+
+&gmac0 {
+	qcom,poll_required = <1>;
+	qcom,poll_required_dynamic = <1>;
+	qcom,phy_mdio_addr = <3>;
+	vlan_tag = <1 0x10>;
+};
+
+&gmac1 {
+	qcom,poll_required = <1>;
+	qcom,poll_required_dynamic = <1>;
+	qcom,phy_mdio_addr = <4>;
+	vlan_tag = <2 0x20>;
+};
+
+&wifi0 {
+	status = "okay";
+	qcom,ath10k-calibration-variant = "8devices-Jalapeno";
```

Comment:
as far as I know, your board.bin wasn't yet accepted upstream yet. If I'm right, please make it a comment. Same for the other wifi node.

Reply:
The board-2.bin in ipq-wifi is already using the variant strings:

    $ strings board-8dev_jalapeno.bin|grep bus
    bus=ahb,bmi-chip-id=0,bmi-board-id=16,variant=8devices-Jalapenom
    bus=ahb,bmi-chip-id=0,bmi-board-id=17,variant=8devices-Jalapenom

So giving it in the DTS is required (and nice do have when it is merged in the official board-2.bin)

Reply:
Thanks a lot for the explanation! Quite obvious that it need to be kept as it is now.